Lib/test/test_threading.py

```diff
@@ -907,6 +907,13 @@ def __call__(self):
         thread.join()
         self.assertTrue(target.ran)
 
+    def test_leak_without_join(self):
+        # bpo-37788: Test that a thread which is not joined explicitly
+        # does not leak. Test written for reference leak checks.
+        def noop(): pass
+        with threading_helper.wait_threads_exit():
+            threading.Thread(target=noop).start()
```

Comment:
Would you mind to add a test case that the target function isn't just only the `pass`

Reply:
Hmm, what do you mean exactly? I'm not sure I understand your suggestion.

Reply:
Oh, sorry. I didn't explain it clearly. I suggest that the target of Thread could be more complex ;)

Something like:
```python
def noop(): 
      # we don't what code will run in the thread in fact.
      time.sleep(random.random())
with threading_helper.wait_threads_exit():
      threading.Thread(target=noop).start()
```

Reply:
But what would that change? What matters is what happens after the thread ends.

Reply:
Hm, give me a second.
No matter how complex of the target of thread, the thread will finish in time in python level. So we don't adding a complex testcase in here is fine. right?

Reply:
Exactly.